src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/seed/cas1/Cas1SeedPremisesFromSiteSurveyXlsxJob.kt

```diff
@@ -0,0 +1,289 @@
+    existingPremise.apply {
+      name = siteSurvey.name
+      addressLine1 = siteSurvey.address
+      addressLine2 = null
+      town = siteSurvey.townCity
+      postcode = siteSurvey.postcode
+      probationRegion = premisesInfo.probationRegion
+      localAuthorityArea = premisesInfo.localAuthorityArea
+      longitude = premisesInfo.longitude
+      latitude = premisesInfo.latitude
+      point = premisesInfo.point
+      gender = siteSurvey.maleFemale.toApprovedPremisesGender()
+    }
+
+    existingPremise.characteristics.clear()
+    existingPremise.characteristics.addAll(premisesInfo.characteristicEntities)
+
+    val afterChange = ApprovedPremisesForComparison.fromEntity(existingPremise)
+
+    val diff = javers.compare(beforeChange, afterChange)
+    if (diff.hasChanges()) {
+      log.info("Changes are ${diff.prettyPrint()}")
+      premisesRepository.save(existingPremise)
+    } else {
+      entityManager.clear()
+      log.info("No changes detected")
+    }
+  }
+
+  private fun MaleFemale.toApprovedPremisesGender() = when (this) {
+    MaleFemale.MALE -> ApprovedPremisesGender.MAN
+    MaleFemale.FEMALE -> ApprovedPremisesGender.WOMAN
+  }
+
+  private data class CharacteristicRequired(
+    val propertyName: String,
+    val value: Boolean,
+  )
+
+  private data class ApprovedPremisesForComparison(
+    val id: UUID,
+    val name: String,
+    val addressLine1: String,
+    val addressLine2: String?,
+    val town: String?,
+    val postcode: String,
+    val longitude: Double?,
+    val latitude: Double?,
+    val notes: String,
+    val emailAddress: String?,
+    val probationRegionName: String,
+    val localAuthorityAreaName: String?,
+    val apCode: String,
+    val qCode: String,
+    val characteristicNames: List<String>,
+    val status: PropertyStatus,
+    val point: Point?,
+    val gender: ApprovedPremisesGender,
+    val supportsSpaceBookings: Boolean,
+    val managerDetails: String?,
+  ) {
+    companion object {
+      fun fromEntity(entity: ApprovedPremisesEntity) = ApprovedPremisesForComparison(
+        id = entity.id,
+        name = entity.name,
+        addressLine1 = entity.addressLine1,
+        addressLine2 = entity.addressLine2,
+        town = entity.town,
+        postcode = entity.postcode,
+        longitude = entity.longitude,
+        latitude = entity.latitude,
+        notes = entity.notes,
+        emailAddress = entity.emailAddress,
+        probationRegionName = entity.probationRegion.name,
+        localAuthorityAreaName = entity.localAuthorityArea?.name,
+        apCode = entity.apCode,
+        qCode = entity.qCode,
+        characteristicNames = entity.characteristics.map { it.name }.sorted(),
+        status = entity.status,
+        point = entity.point,
+        gender = entity.gender,
+        supportsSpaceBookings = entity.supportsSpaceBookings,
+        managerDetails = entity.managerDetails,
+      )
+    }
+  }
+
+  @SuppressWarnings("TooGenericExceptionThrown")
```

Comment:
There is a new `SiteSurveyImportException` class which might be applicable here.

Reply:
I noticed that but i didn't really see the value given the seed service is just catching all exceptions anyway. I find `error("asdasd")` a more fluent way of raising unrecoverable errors.